src/main/java/duke/task/ToDo.java

```diff
@@ -0,0 +1,94 @@
+package duke.task;
+
+import java.util.ArrayList;
+import duke.DukeException;
+
+/**
+ * Represents a ToDo task.
+ */
+public class ToDo extends Task {
+    public String description;
+
+    /**
+     * Constructs a ToDo task with the given description.
+     *
+     * @param description The description of the ToDo task.
+     */
+    public ToDo(String description) {
+        super(description);
+        this.description = description;
+    }
+
+    /**
+     * Returns a string representation of the ToDo task.
+     *
+     * @return A string representation of the ToDo task.
+     */
+    @Override
+    public String toString() {
+        return "[T]" + super.toString();
+    }
+
+    /**
+     * Returns a string representation of the ToDo task for writing to a file.
+     *
+     * @return A string representation of the ToDo task for file output.
+     */
+    public String toStringFile() {
+        return "T | " + super.toStringFile();
+    }
+
+    /**
+     * Gets the type of the task.
+     *
+     * @return The type of the task.
+     */
+    @Override
+    public String getType() {
+        return "TODO";
+    }
+
+    /**
+     * Adds a ToDo task to the list of tasks.
+     *
+     * @param description The description of the ToDo task.
+     * @param list The ArrayList to which the task will be added.
+     * @return The added ToDo task.
+     * @throws DukeException If the description is empty.
+     */
+    public static ToDo addTodo(String description, ArrayList<Task> list) throws DukeException {
```

Comment:
Not sure whether this function and the below one are suitable to be written inside the task class itself... Seems like they are more like the functionalities of TaskList that manages all the tasks inside it.

Reply:
Good point! Might consider moving some of the functions in the future. Thanks for the feedback!